### PR TITLE
Vs 2804 pre show countdown

### PR DESCRIPTION
**Pull request by molnartimi**

### CODE REVIEW

**Review by kecskesk — APPROVED** [review]

File: react/features/filmstrip/components/native/InFocusView.js [hunk 1]
```diff
@@ -176,6 +186,16 @@ class InFocusView extends Component<Props> {
                 </View>}
             </View>);
     }
+
+    _createCountdownIfNeeded() {
+        if (this.props.countdownStartDatetime && this.props.countdownTargetDatetime) {
```
Comment (vetrabmercedes) [inline react/features/filmstrip/components/native/InFocusView.js]: Please use lodash _.isEmpty() method for empty checking.

File: react/features/filmstrip/components/native/PreShowCountdown.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    /**
+     * Check if remaining time is initialized and not finished.
+     *
+     * @param {Moment} time - Remaining time to check.
+     * @private
+     * @returns {boolean}
+     */
+    _isRemainingTimeValid(time): boolean {
+        return time && time.valueOf() >= 0;
```
Comment (vetrabmercedes) [inline react/features/filmstrip/components/native/PreShowCountdown.js:102]: Please use lodash _.isEmpty() or _.isNil() method for nullcheck

**Review by bruderbarna — APPROVED** [review]

File: react/features/filmstrip/components/native/PreShowCountdown.js (hunk 2 above)
Comment (vetrabmercedes) [inline react/features/filmstrip/components/native/PreShowCountdown.js:102]: Another thing after nullcheck: I saw that you install the moment package. Please use the 
moment("not a real date").isValid();
function to validate that is string is a date or not.

File: react/features/filmstrip/components/native/PreShowCountdown.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    _getFormattedTimeString(): string {
+        const days = Math.floor(this.state.remainingTime.valueOf() / 1000 / 60 / 60 / 24);
+        const daysString = days ? `${days} day${days > 1 ? 's' : ''} | ` : '';
+
+        const hours = Math.floor(this.state.remainingTime.valueOf() / 1000 / 60 / 60) % 24;
+        const hoursString = hours || days ? `${hours} hr | ` : '';
+
+        const minAndSecString = this.state.remainingTime.format('mm [min] | ss [sec]');
+
+        return daysString + hoursString + minAndSecString;
```
Comment (vetrabmercedes) [inline react/features/filmstrip/components/native/PreShowCountdown.js:120]: Please use the moment package for dateformatting, for example 
moment("12-25-1995", "MM-DD-YYYY");
You can find more examples here: 
https://momentjs.com/docs/

**Review by vetrabmercedes — APPROVED** [review]

**Pull request merged**